feat(oxc_language_server):  add diagnostic code with link

IWANABETHATGUY opened the pull request:
1. follow #6479 
2. Adding related doc link to lsp diagnostic code.
## Preview
![image](https://github.com/user-attachments/assets/dad5b6a2-787a-43c3-b308-ebe74e38e21c)

**after click**
![image](https://github.com/user-attachments/assets/5b725fcd-cacd-4779-9b2a-15824d7c0cf7)

The UI looks a little wired, due to https://github.com/oxc-project/oxc/blob/aba9ea3f1d68c8b32e2b8aba7d6038737f1652eb/crates/oxc_diagnostics/src/lib.rs#L113-L120, 

I suggest change the code above to 
```rs
format!("{scope}/{number}")
```

DonIsaac commented on crates/oxc_language_server/src/linter.rs:
```diff
@@ -107,7 +110,13 @@ impl ErrorWithPosition {
         );
-        let source = self.miette_err.code().map_or("oxc".to_string(), |code| format!("oxc:{code}"));
+        let code = self.miette_err.code().map(|item| item.to_string());
+        let code_description = code.as_ref().and_then(|code| {
```
If you need a doc link, maybe we add it to OxcDoagnostic directly?

IWANABETHATGUY replied:
Yeah, that should be the most perfect solution. but I'm not sure if adding a new field hurt linter performance.

DonIsaac replied:
CPU tends to be cheaper than memory. It all depends on this: when we provide diagnostics as JSON, do we want to make those links available? This affects use cases in external crates/applications

DonIsaac replied:
Also: please consider this comment non blocking. This PR looks good to me.

IWANABETHATGUY replied:
Ok, let's polish in the future.

IWANABETHATGUY commented:
### Merge activity

* **Oct 13, 1:01 AM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

IWANABETHATGUY committed:
feat(oxc_language_server):  add diagnostic code with link (#6487)

1. follow #6479
2. Adding related doc link to lsp diagnostic code.
## Preview
![image](https://github.com/user-attachments/assets/dad5b6a2-787a-43c3-b308-ebe74e38e21c)

**after click**
![image](https://github.com/user-attachments/assets/5b725fcd-cacd-4779-9b2a-15824d7c0cf7)

The UI looks a little wired, due to https://github.com/oxc-project/oxc/blob/aba9ea3f1d68c8b32e2b8aba7d6038737f1652eb/crates/oxc_diagnostics/src/lib.rs#L113-L120,

I suggest change the code above to
```rs
format!("{scope}/{number}")
```
